src/utils/graph/layout.js

```diff
@@ -39,56 +39,36 @@ export const layout = ({
     node.y = 0;
   }
 
-  // Constants passed to solver
+  // Constants used by constraints
   const constants = {
     spaceX,
     spaceY,
-    basisX,
+    spreadX,
     layerSpace: (spaceY + layerSpaceY) * 0.5,
   };
 
-  // Constraints to separate nodes into rows
+  // Constraints to separate nodes into rows and layers
   const rowConstraints = createRowConstraints(edges);
-
-  // Constraints to separate nodes into layers
   const layerConstraints = createLayerConstraints(nodes, layers);
 
   // Find the node positions given these constraints
   solve([...rowConstraints, ...layerConstraints], constants, 1, true);
 
-  // Find the rows using the node positions after solving
+  // Find the solved rows using the node positions after solving
   const rows = groupByRow(nodes);
 
-  // Constraints to avoid edges crossing
-  const crossingConstraints = createCrossingConstraints(edges);
-
-  // Constraints to maintain parallel vertical edges
-  const {
-    parallelConstraints,
-    parallelSingleConstraints,
-    parallelDoubleConstraints,
-  } = createParallelConstraints(edges);
-
-  // Constraints to maintain a minimum horizontal node spacing
-  const separationConstraints = createSeparationConstraints(rows);
+  // Constraints to avoid edges crossing and maintain parallel vertical edges
+  const crossingConstraints = createCrossingConstraints(edges, constants);
+  const parallelConstraints = createParallelConstraints(edges, constants);
 
-  // Solve these constraints using multiple iterations
+  // Solve these constraints iteratively
   for (let i = 0; i < iterations; i += 1) {
-    // Solve main constraints
     solve(crossingConstraints, constants, 1);
-    solve(parallelConstraints, constants, 1);
-
-    // Further improve special cases with more effort
-    solve(parallelSingleConstraints, constants, iterations * 0.5);
-    solve(parallelDoubleConstraints, constants, iterations * 0.5);
-
-    // Update and solve separation constraints given the updated positions
-    updateSeparationConstraints(separationConstraints, rows, spaceX);
-    solve(separationConstraints, constants, iterations * 0.5);
+    solve(parallelConstraints, constants, 50);
   }
 
-  // Update separation constraints but ensure spacing is exact
-  updateSeparationConstraints(separationConstraints, rows, spaceX, true);
+  // Constraints to maintain a minimum horizontal node spacing
```

Comment:
This change is very interesting! Similar to my comment above, it would be great to document the reasoning / motivation behind this change in the PR description.

Reply:
Ah this part was just a refactoring and simplification so I've not much to add on that.